rero_ils/modules/loans/listener.py

```diff
@@ -41,10 +47,62 @@ def enrich_loan_data(sender, json=None, record=None, index=None,
 def listener_loan_state_changed(_, prev_loan, loan, trigger):
     """Create notification based on loan state changes."""
     if loan.get('state') == 'PENDING':
+        # create notification to requester
         item_pid = loan.get('item_pid')
         checkedout_loan_pid = Item.get_loan_pid_with_item_on_loan(item_pid)
         if checkedout_loan_pid:
             checked_out_loan = Loan.get_record_by_pid(checkedout_loan_pid)
             checked_out_loan.create_notification(notification_type='recall')
+        # send notification to location if needed
+        item = Item.get_record_by_pid(item_pid)
+        item_location = Location.get_record_by_pid(item.location_pid)
+        if item_location and item_location.get('send_notification', False):
+            send_notification_to_location(loan, item, item_location)
     elif loan.get('state') == 'ITEM_AT_DESK':
         loan.create_notification(notification_type='availability')
+
+
+def check_loan_allow_request(sender, item=None, reasons=[], **kwargs):
+    """Signal sent to check if an item can be requested.
+
+    :param sender: signal sender function
+    :param item: the item to check
+    :param reasons: a list of reasons why an item could ne be requested
+    :param kwargs: additional keyword arguments
+    """
+    # if required arguments aren't available. No need to continue
+    if None in [item, kwargs.get('patron_barcode')]:
+        return
```

Comment:
Why not returning reasons?

Reply:
If these arguments are not present. This 'hook' is not relevant and mut be not evaluated.